### ISSUE TITLE
boinc client crashed at start

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#749]
**Reported by AngieSoft on 11 Jul 38863801 00:00 UTC**
the client crashed after computing astropulse 4.36 after restart with the follow protocol

**Comment by romw** [comment]
**Commented by Nicolas on 5 Aug 38863858 21:46 UTC**
Looks like it closed an invalid or already-closed ```FILE*``` (```fclose``` is mentioned in stack trace). I'll try to decode the other stacktrace items.

**Comment by romw** [comment]
**Commented by Nicolas on 14 Feb 38863962 09:46 UTC**
```
./boinc[TIME_STATS::get_log_after(double, MIOFILE&)    /home/boincadm/src/boinc_6.2/client/time_stats.C:144
./boinc[0x42925d](0x446c82]) CLIENT_STATE::make_scheduler_request(PROJECT*) /home/boincadm/src/boinc_6.2/client/cs_scheduler.C:274
./boinc[SCHEDULER_OP::init_op_project(PROJECT*, int)   /home/boincadm/src/boinc_6.2/client/scheduler_op.C:135
./boinc[0x428a02](0x43dcad]) CLIENT_STATE::scheduler_rpc_poll()             /home/boincadm/src/boinc_6.2/client/cs_scheduler.C:347
./boinc[CLIENT_STATE::poll_slow_events()               /home/boincadm/src/boinc_6.2/client/client_state.C:588
./boinc[0x43b71d](0x41699b]) boinc_main_loop()                              /home/boincadm/src/boinc_6.2/client/main.C:472
```

**Comment by romw** [comment]
**Commented by Nicolas on 1 Feb 38863973 00:53 UTC**
It's all in the ```TIME_STATS``` class.

If the ```fopen``` call in the ```start()``` method fails, the error is silently ignored, leaving ```time_stats_log``` set to ```NULL```. ```log_append()``` checks if the pointer is valid before trying to call ```fprintf```, so that method doesn't crash. However, ```get_log_after()``` doesn't do any check, it calls ```fclose``` on a potentially-NULL pointer.

**Comment by romw** [comment]
**Commented by davea on 25 May 38866460 06:13 UTC**
(In [16478]) - client: don't fclose() time_stats_log if it's NULL
    Fixes #772
